File: basex/lua/read/basex_read.lua
#! /usr/bin/lua
-- ----------------------------------------------------
--	basex_read.lua
--
--					Feb/02/2012
-- ----------------------------------------------------
require ('LuaXml')
require ('xml_manipulate')

-- ----------------------------------------------------
print ("*** 開始 ***")

local http = require ("socket.http")

local resp = {}

local rr,code,header = http.request{
	url = "http://admin:admin@localhost:8984/rest/cities?query=/",
	sink = ltn12.sink.table( resp ),
}

xml_str = table.concat (resp)
-- print (xml_str)

xml_string_display (xml_str)

print ("*** 終了 ***")
-- ----------------------------------------------------

File: basex/ocaml/read/basex_read.ml
(* ------------------------------------------------------------------- *)
(*
	basex_read.ml

					Feb/13/2012

*)
(* ------------------------------------------------------------------- *)
open	Xml
open	Xml_manipulate
open	Curl_manipulate
(* ------------------------------------------------------------------- *)
let _ = print_endline "*** 開始 ***"
let url_target = "http://admin:admin@localhost:8984/rest/cities?query=/"

let _ = curl_get_proc url_target show_xml_str_proc

let _ = print_endline "*** 終了 ***"
(* ------------------------------------------------------------------- *)
